test/stage2/test.zig

```diff
@@ -938,12 +938,25 @@ pub fn addCases(ctx: *TestContext) !void {
         \\pub fn b() !void {
         \\	defer return a();
         \\}
-
-     , &[_][]const u8{
+    , &[_][]const u8{
         ":7:8: error: try is not allowed inside defer expression",
         ":10:8: error: cannot return from defer expression",
     });
 
+    ctx.compileError("ambigious references", linux_x64,
+        \\const T = struct {
+        \\    const T = struct {
+        \\        fn f() void {
+        \\            _ = T;
+        \\        }
+        \\    };
+        \\};
+    , &.{
+        ":4:17: error: ambigious reference",
+        ":1:1: note: first reference here",
+        ":2:5: note: second reference here",
```

Comment:
A reference is what is happening on line 4; these are declarations:

```suggestion
        ":1:1: note: declared here",
        ":2:5: note: also declared here",
```